module #3. REST API Advanced/core/src/main/java/com/epam/esm/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,49 @@
+package com.epam.esm.service.impl;
+
+import com.epam.esm.dao.OrderDao;
+import com.epam.esm.dto.GiftCertificateDto;
+import com.epam.esm.dto.OrderDto;
+import com.epam.esm.entity.Order;
+import com.epam.esm.service.GiftCertificateService;
+import com.epam.esm.service.OrderService;
+import com.epam.esm.service.mapper.OrderDtoMapper;
+import org.springframework.stereotype.Service;
+
+import java.time.ZonedDateTime;
+import java.util.List;
+
+@Service
+public class OrderServiceImpl implements OrderService {
+    private final OrderDao orderDao;
+    private final OrderDtoMapper orderDtoMapper;
+    private final GiftCertificateService giftCertificateService;
+
+    public OrderServiceImpl(OrderDao orderDao, OrderDtoMapper orderDtoMapper,
+                            GiftCertificateService giftCertificateService){
+        this.giftCertificateService = giftCertificateService;
+        this.orderDao = orderDao;
+        this.orderDtoMapper = orderDtoMapper;
+    }
+
+    @Override
+    public long add(OrderDto orderDto) {
+        Order order = orderDtoMapper.toEntity(orderDto);
+        GiftCertificateDto certificateDto = giftCertificateService.getById(order.getCertificateId());
```

Comment:
it's ok, or better to create separate method for certificate "getCertificatePriceById(long id)" than query full entity just for one field value?